### PR TITLE
[INF-565] fix: render ruby in Text Reader page and button labels

### PR DESCRIPTION
**Pull request by KirylHatalski**
## Summary
- Stop HTML-escaping Text Reader page/nav labels and safely render ruby from `{ruby}` placeholders / ruby HTML in Previous/Next button labels ([INF-565](https://oat-sa.atlassian.net/browse/INF-565)).
- Bump Text Reader IMS PCI to `1.3.5` with a migration to re-register the portable element.
- Includes a temporary en-US ruby annotation on `Page` for test-env verification.

chrome_uMAt4YQqoA

## Test plan
- [ ] Run extension update / migration so Text Reader IMS PCI `1.3.5` is registered
- [ ] Create a **new** Text Reader item on en-US: `Page` shows furigana `test`
- [ ] Set Button labels to `{ruby}{rb}前{/rb}{rt}mae{/rt}{/ruby}` — furigana renders
- [ ] Set Button labels to `<ruby>前<rt>mae</rt></ruby>` — furigana renders
- [ ] Set Button labels to `` — no alert / markup stripped or escaped
- [ ] Remove the temporary en-US `Page` test annotation before release if not needed

[INF-565]: https://oat-sa.atlassian.net/browse/INF-565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Text Reader navigation labels now support ruby annotations and selected, safely rendered HTML.
  * Previous/next button labels and page indicators display localized content consistently across supported versions.
* **Bug Fixes**
  * Improved handling of missing, empty, malformed, or non-text labels.
  * Prevented unsafe markup from being rendered in navigation controls.
* **Chores**
  * Updated the PCI TextReader IMS client to version 1.3.5.
* **Tests**
  * Added coverage for label rendering, sanitization, escaping, and edge cases across both Text Reader versions.

### CODE REVIEW

**Comment by KirylHatalski** [comment]
@coderabbitai review

File: locales/en-US/messages.po [hunk 1]
```diff
@@ -56,7 +56,7 @@ msgid "Next"
 msgstr ""
 
 msgid "Page"
-msgstr ""
+msgstr "{ruby}{rb}Page{/rb}{rt}test{/rt}{/ruby}"
```
[the remark this answers is not in the document]
Comment (KirylHatalski) [reply locales/en-US/messages.po]: required for QA test, will be removed before merge

File: views/js/pciCreator/dev/textReaderInteraction/runtime/js/buttonLabel.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        if (typeof label !== 'string') {
+            return escapeHtml(label);
+        }
+
+        if (label === '') {
+            return '';
+        }
+
+        return sanitizeRubyHtml(convertRubyTags(label));
+    }
```
[the remark this answers is not in the document]
Comment (KirylHatalski) [reply views/js/pciCreator/dev/textReaderInteraction/runtime/js/buttonLabel.js:113]: UT added

**Review by bartlomiejmarszal — APPROVED** [review]

File: views/js/pciCreator/ims/textReaderInteraction/imsPciCreator.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
         "src": [
             "./runtime/textReaderInteraction.js",
             "./runtime/js/renderer.js",
+            "./runtime/js/buttonLabel.js",
```
Comment (Karol-Stelmaczonek) [inline views/js/pciCreator/ims/textReaderInteraction/imsPciCreator.json:25]: I think we need to add similar declaration to tao/pciSamples/views/js/pciCreator/dev/textReaderInteraction/pciCreator.json

File: locales/en-US/messages.po (hunk 1 above)
Comment (Karol-Stelmaczonek) [inline locales/en-US/messages.po]: - [x] remove the test value before merge

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: views/js/pciCreator/ims/textReaderInteraction/imsPciCreator.json (hunk 3 above)
Comment (KirylHatalski) [reply views/js/pciCreator/ims/textReaderInteraction/imsPciCreator.json:25]: done
